src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/LimitedUseItem.java

```diff
@@ -0,0 +1,151 @@
+     */
+    public final int getMaxUseCount() {
+        return maxUseCount;
+    }
+
+    /**
+     * Sets the maximum number of times this item can be used.
+     * The number must be greater than zero.
+     *
+     * @param count The maximum number of times this item can be used.
+     * @return The {@link LimitedUseItem} for chaining of setters
+     */
+    public final @Nonnull LimitedUseItem setMaxUseCount(int count) {
+        Validate.isTrue(count > 0, "The maximum use count must be greater than zero!");
+
+        maxUseCount = count;
+        return this;
+    }
+
+    /**
+     * Returns the {@link NamespacedKey} under which will the amount of uses left stored.
+     *
+     * @return The {@link NamespacedKey} to store/load the amount of uses
+     */
+    protected @Nonnull NamespacedKey getStorageKey() {
+        return defaultUsageKey;
+    }
+
+    @Override
+    public void register(@Nonnull SlimefunAddon addon) {
+        if (getMaxUseCount() < 1) {
+            warn("The use count has not been configured correctly. It needs to be at least 1. The Item was disabled.");
+        } else {
+            super.register(addon);
+        }
+    }
+
+    @ParametersAreNonnullByDefault
+    protected void damageItem(Player p, ItemStack item) {
+        if (item.getAmount() > 1) {
+            item.setAmount(item.getAmount() - 1);
+
+            // Separate one item from the stack and damage it
+            ItemStack separateItem = item.clone();
+            separateItem.setAmount(1);
+            damageItem(p, separateItem);
+
+            // Try to give the Player the new item
+            if (!p.getInventory().addItem(separateItem).isEmpty()) {
+                // or throw it on the ground
+                p.getWorld().dropItemNaturally(p.getLocation(), separateItem);
+            }
+        } else {
+            ItemMeta meta = item.getItemMeta();
+            NamespacedKey key = getStorageKey();
+            PersistentDataContainer pdc = meta.getPersistentDataContainer();
+            int usesLeft = pdc.getOrDefault(key, PersistentDataType.INTEGER, getMaxUseCount());
+
+            if (usesLeft == 1) {
+                p.playSound(p.getLocation(), Sound.ENTITY_ITEM_BREAK, 1, 1);
+                item.setAmount(0);
+                item.setType(Material.AIR);
+            } else {
+                usesLeft--;
+                pdc.set(key, PersistentDataType.INTEGER, usesLeft);
+
+                updateItemLore(item, meta, usesLeft);
+            }
+        }
+    }
+
+    @ParametersAreNonnullByDefault
+    private void updateItemLore(ItemStack item, ItemMeta meta, int usesLeft) {
+        List<String> lore = meta.getLore();
+
+        String newLine = ChatColors.color(LoreBuilder.usesLeft(usesLeft));
+        if (lore != null && !lore.isEmpty()) {
+            // find the correct line
+            for (int i = 0; i < lore.size(); i++) {
+                if (PatternUtils.USES_LEFT_LORE.matcher(lore.get(i)).matches()) {
```

Comment:
Do we want a regex for this? we could just as easily .endsWith("Uses left")

Reply:
Good point, but since we're discussing, how would this get translated in the future? :smile:

Reply:
Easily if it's ends with, not as easily with regex :p

Reply:
Just to clarify, I didn't imply that it would be easier with regex, I just didn't think of a way to do it with endsWith :smile: (in the one minute I thought about it)